src/main/java/li/cil/tis3d/common/network/handler/AbstractMessageHandler.java

```diff
@@ -28,44 +28,6 @@ public void onMessage(final T message, final PacketContext context) {
 
     @Nullable
     protected World getWorld(final DimensionType dimension, final PacketContext context) {
-        switch (context.getPacketEnvironment()) {
-            case CLIENT:
-                //noinspection MethodCallSideOnly Guarded by CLIENT switch case.
-                return getWorldClient(dimension, context);
-            case SERVER:
-                return getWorldServer(dimension, context);
-            default:
-                return null;
-        }
-    }
-
-    @Environment(EnvType.CLIENT)
-    @Nullable
-    private static World getWorldClient(final DimensionType dimension, final PacketContext context) {
-        final PlayerEntity player = context.getPlayer();
-        if (player == null) {
-            return null;
-        }
-
-        final World world = player.world;
-        if (world == null) {
-            return null;
-        }
-
-        if (world.getDimension().getType() != dimension) {
-            return null;
-        }
-
-        return world;
-    }
-
-    @Nullable
-    private static World getWorldServer(final DimensionType dimension, final PacketContext context) {
-        final MinecraftServer server = context.getPlayer().getServer();
-        if (server == null) {
-            return null;
-        }
-
-        return server.getWorld(dimension);
+        return context.getPlayer().world;
```

Comment:
My current understanding is as follow. If all queued server->client update messages go through this method or one of the similar ones:
```java
public int sendToClientsInDimension(final AbstractMessage message, final World world) {
       // [...]
        int sent = 0;
        for (final PlayerEntity player : world.getPlayers()) {
            if (player instanceof ServerPlayerEntity) {
                ((ServerPlayerEntity)player).networkHandler.sendPacket(packet);
                sent++;
            }
        }

        return sent;
    }
```

then a problem could only arise if this happens from a different thread than the main one, because at this point, the player is either still in the old dimension (and is thus caught in the iteration), or isn't and gets ignored. I say this though while having a very shallow understanding of MC dimension and networking code; all I know is that it uses TCP and network requests are (hopefully) all handled on the same thread as world state updates :)

Reply:
Thinking really hard of a way to force the scenario of "packet sent after dimension hop" :thinking:

Reply:
Comment from UpcraftLP discord:
> 1) yes, mc packets are handled in the same order as they are sent. so if you iterate over all players in a dimension, you'll get all that are in there, they will not be in some flux state during teleportation.
> 2) if you're doing that, just use fabric api's player stream thingy

Reply:
Ah, your argument re: iterating players in world is what made it click for me. I was kinda hung up on the casing queues potentially persisting several frames due to throttling and worlds getting unloaded in the meantime. But then those worlds would not contain any players, so it really doesn't matter!

OK, in that case I don't really have any more concerns regarding that change. Thanks for looking into this for me!

Only thing I'll need to sleep over some more is whether to kill off the `AbstractMessageWithDimension` type in the future, since leaving it in just as a "marker" seems somewhat odd to me. But that's something for another day.